src/generator.py

```diff
@@ -72,9 +72,9 @@ async def send_request(
                 retry_delay = min(retry_delay * 2, max_retry_delay)  # Exponential backoff
 
             except APIStatusError as e:
-                # Handle specific API status codes, like 503 Service Unavailable, for retries
-                if e.status_code == 503:
-                    logging.warning(f"Service unavailable (attempt {attempt + 1}/{max_retries}): {e}. Retrying in {retry_delay:.2f}s...")
+                # Handle specific API status codes for retries, like 429 (rate limits) and 503 (availability)
```

Comment:
In the case of rate limits, there are actually some hints embedded in the response headers: https://inference-docs.cerebras.ai/support/rate-limits

So if you hit a rate limit you will be able to know precisely how long you need to wait.